NSS-Day-Cohort-42/rare-API-media-moguls
# User profile details retrieved for single user.

#### Changes Made
1. Modified file `user.py` in `views` to include `retrieve` functionality to get single user.
​1. Modified file pull request template to simplify and streamline template.
1. Removed duplicate import from file `urls.py`
1. Modified fixtures `rareusers.json` and `users.json` to include an active admin, an active author, and an inactive author in user db.
1. Included descriptive doc strings on unmapped properties and added `is_current_user` property in `rareuser.py` model.
1. Added doc strings to `__init__.py` and `auth.py` in views.
1. Imported `CategorySerializer` to `post.py` in views & added `id` to fields on `PostRareUserSerializer` 

#### Steps to Review
1. Checkout this branch locally.
```
git fetch --all
git checkout mp-user-profile
```
2. Open a new Terminal tab (⌘T) and navigate to the server directory.
3. Initialize virtual environment, and run the server.
```
pipenv shell
python manage.py runserver
```
4. Nuke your db to access updated database
```
./nuke.sh
```
5. Test app functionality.
> Create a GET request in Postman with URL `http://localhost:8000/users/2` and auth token `fa2eba9be8282d595c997ee5cd49f2ed31f65bed` (for user 1)
> Expected return:
```
{
    "id": 2,
    "bio": "World renowned book enthusiast, instantgram influencer, and ticktockticktockticktock kween Theo Daly began building her empire on the popular social media platform SnapCrackleChat in 2001. With a net worth of $8.1 bazillion, Forbey Parker ranked her somewhere in the middle of their list of the world’s kajillionaires.",
    "is_staff": false,
    "is_active": true,
    "full_name": "Theo Daly",
    "profile_image_url": null,
    "is_current_user": false,
    "username": "theocanread",
    "email": "theo@daly.com",
    "date_joined": "2020-08-28T14:51:39.989000Z"
}
```
> Create a GET request in Postman with URL `http://localhost:8000/users/2` and auth token `fa2eba9be8282d595c997ee5cd49f2ed31f65bed` (for user 1)
> Expected return:
```
{
    "id": 1,
    "bio": "Internationally acclaimed scuba instructor, radio dj personality, and mid-century modern sofa collector, Mary Mogul is often referred to as 'Awesome', 'Super Fantastic', and 'Great'. Her talk show 'Mary The Mogul' ran from 2019 to 2020 and made history by becoming the most 'wow' program of its kind ever to be aired, and the most 'woah' program of its kind ever to be cancelled. With a net worth of $3.1 gazillion, Forbey Parker ranked her at the top of their list of the world’s kajillionaires.",
    "is_staff": true,
    "is_active": true,
    "full_name": "Mary Mogul",
    "profile_image_url": null,
    "is_current_user": true,
    "username": "marymogul",
    "email": "mary@mogul.com",
    "date_joined": "2020-08-28T14:51:39.989000Z"
}
```
6. View code file.
> Confirm file modifications are present as indicated above.
> Confirm no unused code or extraneous comments exist.
